BP-19: Update WordPress plugin landing page

gitstart-blockprotocol opened the pull request:
🌟 What is the purpose of this PR?
---
New Design for WordPress page

---
🔗  Related links
---
- [Issue Link](https://github.com/blockprotocol/blockprotocol/issues/1224)

❓How to test this?
---

https://www.loom.com/share/410631370346464c8a8642b10e071cbc

---
This code was written and reviewed by GitStart Community. Growing future engineers, one PR at a time.

gitstart-blockprotocol commented on apps/site/src/components/pages/wordpress/header.tsx:
```diff
@@ -1,16 +1,18 @@
   const theme = useTheme();
 
-  const [loading, setLoading] = useState(true);
+  const onInstallWordPress = () => {
+    // TODO
+  };
+
+  const onDownload = () => {
+    // TODO
+  };
```
@nonparibus, These functions as the name describes are meant to be called when the individual buttons are clicked, a `// TODO` is placed because we do not know the functionality details and how you'd want it implemented

gitstart-blockprotocol commented on apps/site/src/components/pages/wordpress/details.tsx:
```diff
@@ -0,0 +1,530 @@
+              }}
+            >
+              <CustomLinkButton
+                href="/"
```
@nonparibus, As well as the `href`. We did not want to imply the link so we're bringing it to your notice

nonparibus reviewed (changes requested):
Dropped a few first-pass initial comments on the Vercel deployments @ https://blockprotocol-git-bp-oss-1224.stage.hash.ai/wordpress. The produced site has only been tested on desktop at this point. If you can't view the Vercel comments, try logging in to a Vercel account. If you still can't view the comments, let us know and we'll explore how best to share this feedback otherwise. We won't be able to add you to our Vercel team, so we may in that case have to find another way. The privacy settings have been changed to allow general public commenting by anybody with the link, but we're unsure whether this exposes existing historical comments, too.

nonparibus commented:
@gitstart-blockprotocol Hi there, we're in a position to review this again. Thanks for your patience. Were you able to access the Vercel deployment comments that were left, regarding changes to be made on this PR? Cheers!

gitstart-blockprotocol commented:
> @gitstart-blockprotocol Hi there, we're in a position to review this again. Thanks for your patience. Were you able to access the Vercel deployment comments that were left, regarding changes to be made on this PR? Cheers!

Hello @nonparibus, Nice to have you guys back. The reviews have been handled and are undergoing internal QA. Will push the updates as soon as it is completed. ETA Thursday

nonparibus commented:
@gitstart-blockprotocol Awesome, thank you!

gitstart-blockprotocol commented:
@nonparibus, a few notes to take, the images on the WordPress page are not high-resolution. This is due to us having to make a trade-off, adding the `svg` directly to the component results in high-resolution images but it makes the component large so we rendered it in the next `Image` component

nonparibus commented:
@gitstart-blockprotocol There are a lot of unresolved Vercel comments still remaining. Please address these fully before re-requesting review. Please comment on those comment threads when you think they have been and addressed, and I will mark them as complete when they have been accepted, so we can track.

nonparibus commented:
There are still multiple pending, unaddressed Vercel comments. We'll re-review when resolved.

gitstart-blockprotocol commented:
> @gitstart-blockprotocol There are still multiple pending, unaddressed Vercel comments. We'll re-review when resolved.

@nonparibus, you were tagged on Vercel on the unaddressed comments now addressed

nonparibus commented:
@LuisChiej Thanks for the follow-up push. I've resolved the comments in Vercel that I think have been addressed (which is all of them) and opened a few more regarding links that hadn't yet been stitched up to their correct locations (or in one case, under the FAQ accordion, were missing). Other than that this is now functionally approved, so tagging @CiaranMn for review here. :) Cheers!

gitstart-blockprotocol commented:
Thanks @nonparibus,
We've addressed the highlighted Vercel comments now, kindly take a look.

nonparibus commented:
Hi @gitstart-blockprotocol, thanks again for your work on this. All existing comments barring three have been accepted as addressed, and no new ones.

The existing ones:
- We still [have issues](https://blockprotocol-jkff955s5.stage.hash.ai/wordpress?vercelThreadId=WNjC0) at certain screen sizes.
- Another comment was, I think, missed (buried under the "What are some example Þ blocks?" accordion item which will need expanding to view it; [direct link also here](https://blockprotocol-jkff955s5.stage.hash.ai/wordpress?vercelThreadId=QcVd6)).
- Third third comment we have [agreed to defer](https://blockprotocol-jkff955s5.stage.hash.ai/wordpress?vercelThreadId=DNgid) looking at until code review.

gitstart-blockprotocol commented:
Hi @nonparibus,

>We still [have issues](https://blockprotocol-jkff955s5.stage.hash.ai/wordpress?vercelThreadId=WNjC0) at certain screen sizes.

We have resolve this now, and also added some notes on our [resolution strategy here](https://blockprotocol-jkff955s5.stage.hash.ai/wordpress?vercelThreadId=WNjC0), would love to get your thoughts on it.

-----
> Another comment was, I think, missed (buried under the "What are some example Þ blocks?" accordion item which will need expanding to view it; [direct link also here](https://blockprotocol-jkff955s5.stage.hash.ai/wordpress?vercelThreadId=QcVd6)).

This is also resolved now, sorry they were missed in initial review 🙏.

------

>Third third comment we have [agreed to defer](https://blockprotocol-jkff955s5.stage.hash.ai/wordpress?vercelThreadId=DNgid) looking at until code review.

We have reverted back to original implementation logic for this, awaiting some further thoughts on it during code review 


Thanks again for looking through this 🙌.

CiaranMn reviewed (commented):
A couple of missing links, otherwise code looks good

nonparibus reviewed (approved):
This is good to merge  🎉

Thanks all for shepherding this through the review process (and for putting up with the long gap in the middle). We'll address the outstanding 2x Vercel comments (both of which I'd missed as well, last time around) separately in follow-up.

Cheers again for all your work on this, and good job!  💯
